bin/link_check.py

```diff
@@ -39,14 +43,15 @@ def test_ftp(url):
         req = urllib.request.Request(url)
         if proxy:
             req.set_proxy(proxy, 'http')
-        response = urlopen(req, timeout=30)
+        response = urlopen(req, timeout=URL_TIMEOUT)
         chunk = response.read(16)
         if len(chunk) == 16:
             res.status_code = 200
         else:
             res.status_code = 404
     except:
-        res.status_code = 404
+        print('ftp exception', url)
+        return Exception()
```

Comment:
There are a few possible reasons to cause exception such as timeout, NIC link down, proxy rejection, proxy timeout, etc. I am not sure if I can list all of them. For those exceptions, we still can not tell if the link is bad or not ( it might be good from outside network), hence pointless to go further to distinguish them. It might be good to just stop at this level and leave it for further manual check.